client/data/reader/use-login-window.ts

```diff
@@ -0,0 +1,78 @@
+import config from '@automattic/calypso-config';
+import { createAccountUrl } from 'calypso/lib/paths';
+
+interface UseLoginWindowProps {
+	onLoginSuccess: () => void;
+}
+
+interface UseLoginWindowReturn {
+	login: () => void;
+	createAccount: () => void;
+}
+
+export default function useLoginWindow( {
+	onLoginSuccess,
+}: UseLoginWindowProps ): UseLoginWindowReturn {
+	const isBrowser: boolean = typeof window !== 'undefined';
+	const environment = config( 'env_id' );
+	let domain = 'wordpress.com';
+	let redirectTo = encodeURIComponent(
+		`https://${ domain }/public.api/connect/?action=verify&service=wordpress`
+	);
+	if ( environment === 'development' ) {
+		domain = 'wpcalypso.wordpress.com';
+		redirectTo = encodeURIComponent(
+			`https://${ domain }/public.api/connect/?action=verify&service=wordpress&domain=${ domain }&origin=${ new URL(
+				window.location.href
+			)?.hostname }`
+		);
+	}
+
+	const loginURL = `https://wordpress.com/log-in?redirect_to=${ redirectTo }`;
```

Comment:
We seem to account for the environment for other URLs, but not for the login and create account URLs. That way, if we start in `development`, we'll end up in staging, which I'm not sure is the expected behavior.